language/move-lang/src/parser/syntax.rs

```diff
@@ -2288,7 +2312,12 @@ fn parse_spec_pragma(tokens: &mut Lexer<'_>) -> Result<SpecBlockMember, Error> {
 //    SpecPragmaProperty = <Identifier> ( "=" Value | ModuleAccess )?
 fn parse_spec_property(tokens: &mut Lexer<'_>) -> Result<PragmaProperty, Error> {
     let start_loc = tokens.start_loc();
-    let name = parse_identifier(tokens)?;
+    let name = match consume_optional_token_with_loc(tokens, Tok::Friend)? {
+        // special treatment for `pragma friend = ...` as friend is a keyword
+        // TODO: this might violate the assumption that a keyword can never be a name.
```

Comment:
Do we need to keyword friend at all?

I might be okay with not doing so

Reply:
Granted I do think it has some nice texture and consistency but we could not do it if we wanted
cc @bob-wilson

Reply:
I tend to favor having `friend` as a keyword, similar to
- `use` in the declarations section and
- `public(script)` in the modifier.

But I definitely would like to hear more thoughts.

Reply:
I think because it is a top level module member, it will be beneficial to have it as a keyword. You really want those things to be keyworded (even if they don't have to be) just for readability/sanity 

I was more highlighting it is an option, since you can check for an identifier and then check the contents

Reply:
I agree about "friend" being a keyword, not because of its use here but because of its use at the top-level module level.

We'll be getting rid of "pragma friend" once the real friend compiler support is done, right?

Reply:
Yes we will get rid of it, but it will need to be incrementally. If its an issue we can also temporary rename from `pragma friend` to `pragma friend_pragma` or such, that should be relative easy.

Reply:
I think it's fine for now. This special case is a bit ugly, but I don't care if it's temporary.